### PR TITLE
changes to support python3 in mgmt-framework

### PR DESCRIPTION
**Pull request by faraazbrcm**
**- Why I did it**

**- How I did it**

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by faraazbrcm** [comment]
@sachinholla @anand-kumar-subramanian please review the changes

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2 [hunk 1]
```diff
@@ -17,6 +17,14 @@ RUN pip install connexion==1.1.15 \
 		six==1.11.0 \
 		urllib3==1.21.1
 
+RUN pip3 install connexion==2.7.0 \
+		setuptools==21.0.0 \
+		grpcio-tools==1.20.0 \
+		certifi==2017.4.17 \
+		python-dateutil==2.6.0 \
+		six==1.11.0 \
+		urllib3==1.21.1
```
Comment (jleveque) [inline dockers/docker-sonic-mgmt-framework/Dockerfile.j2:26]: Suggest to replace tabs with spaces and left-align these lines with `connexion`. Also, please do the same for lines 13-18.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2 (hunk 1 above)
Comment (faraazbrcm) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2]: Thanks for review. I have taken care now

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by lguohan** [comment]
retest vsimage please

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2 [hunk 2]
```diff
@@ -7,15 +7,23 @@ RUN [ -f /etc/rsyslog.conf ] && sed -ri "s/%syslogtag%/$docker_container_name#%s
 ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get update && \
-    apt-get install -y g++ python-dev libxml2
+    apt-get install -y g++ python-dev python3-dev libxml2
```
Comment (jleveque) [inline dockers/docker-sonic-mgmt-framework/Dockerfile.j2:10]: You can probably also remove `python3-dev` on line 43, as it is most likely not needed after we install the Python packages. Just help keep the container size down a bit.

**Review by jleveque — CHANGES_REQUESTED** [review]

**Comment by lguohan** [comment]
@ben-gale , there is not much activity on this one, can you check?

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2 (hunk 2 above)
Comment (faraazbrcm) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2]: @jleveque please review again. I have addressed the review comments

**Comment by faraazbrcm** [comment]
Retest vsimage please

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2 [hunk 3]
```diff
@@ -10,12 +10,20 @@ RUN apt-get update && \
     apt-get install -y g++ python-dev libxml2
 
 RUN pip install connexion==1.1.15 \
```
Comment (jleveque) [inline dockers/docker-sonic-mgmt-framework/Dockerfile.j2:12]: Suggest replacing `pip` with `pip2` to ensure the Python 2 version is used here.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: dockers/docker-sonic-mgmt-framework/Dockerfile.j2 (hunk 2 above)
Comment (jleveque) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2]: So you don't need to install the package at all? I was suggesting to remove the package during cleanup on line 43, but if you have proven that it doesn't need to be installed at all, this is better.

Comment (faraazbrcm) [reply dockers/docker-sonic-mgmt-framework/Dockerfile.j2:10]: Discussed with other contributors, it seems it's needed. Added it back. Addressed pip to pip2 comments. please re-review. thanks, in-advance

**Review by jleveque — APPROVED** [review]

**Pull request merged**